### PR TITLE
Issue #51265 (issue in zoo_1_1 map i.e, salt and fresh water should not coexist).

### LINKED ISSUE #51265: Not really a glitch but there's salt water for the Moose in the Zoo
### Describe the bug

I don't think it makes much sense for it to be salt water. And sorry if this is not the place for this, i wanted to give feedback so i assume here.

### Steps To Reproduce

Just try press E on the shalow water

### Expected behavior

It is salt water

### Screenshots

Screenshot_5

### Versions and configuration

- OS: Windows
    - OS windows 10
- Game Version: 0447a46 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Monsters [No_Fungi]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ps-19**
#### Summary
Mods "[DinoMod] Don't mix salt and fresh water"

#### Purpose of change
Fixes #51265

#### Describe the solution
Fresh water rarely exists next to salt water in nature, and it wouldn't in this situation. So it need to be changed.

#### Describe alternatives you've considered
None

#### Testing
None

#### Additional context
None

### CODE REVIEW

**Comment by ps-19** [comment]
My first contribution in Cataclysm-DDA.

**Review by Maleclypse — APPROVED** [review]

**Comment by actual-nh** [comment]
In order to pass the PR Validator test: for the "Summary", instead of `Category "Bugfixes"` it should be either `None` (which I would recommend) or `Mods "[DinoMod] Don't mix salt and fresh water"`. (These lines are used to put together an automatic first draft of the changelog, so the syntax is somewhat rigid.)

**Comment by ps-19** [comment]
> In order to pass the PR Validator test: for the "Summary", instead of `Category "Bugfixes"` it should be either `None` (which I would recommend) or `Mods "[DinoMod] Don't mix salt and fresh water"`. (These lines are used to put together an automatic first draft of the changelog, so the syntax is somewhat rigid.)

Sir it is done.

**Comment by Zireael07** [comment]
Tip no. 1: Please give the PR a more descriptive title
Tip no 2: Salt and clean water can coexist IRL, google 'haloclines'

**Comment by ps-19** [comment]
> Tip no. 1: Please give the PR a more descriptive title
> Tip no 2: Salt and clean water can coexist IRL, google 'haloclines'

Tip no 1: Done
Tip no 2: Sir my intention is to resolve the issues. Issue maker should be detailed about haloclines, whatever community said i coded accordingly. Owing to your opinion issue #51265 is invalid then.

Quest: Should i close the PR.

**Comment by PatrikLundell** [comment]
I believe tip no 2 by @Zireael07 is simply a comment that the description text about the impossibility for salt and fresh water to coexist is incorrect, not that the issue you're fixing shouldn't be fixed. The zoo situation is not one where water of different salinity would remain separated.

If you want to be precise, you could change the description to say that fresh water rarely exists next to salt water in nature, and it wouldn't in this situation.

Note that I don't have any official position on the project, so this is just an opinion without any weight.

**Comment by ps-19** [comment]
> I believe tip no 2 by @Zireael07 is simply a comment that the description text about the impossibility for salt and fresh water to coexist is incorrect, not that the issue you're fixing shouldn't be fixed. The zoo situation is not one where water of different salinity would remain separated.
> 
> If you want to be precise, you could change the description to say that fresh water rarely exists next to salt water in nature, and it wouldn't in this situation.
> 
> Note that I don't have any official position on the project, so this is just an opinion without any weight.

@PatrikLundell i agree. Thank you.

**Comment by RandDenner** [comment]
@ps-19 

Two things:
1) `Fixes issue #51265` in your summary here (under "purpose of change) should be changed to just `Fixes #51265` in order to have them linked so a merge of this PR closes the issue as well.

2) As I mentioned in the issue, `t_water_dp` and `t_water_sh` are coded with the `"FISHABLE"` tag, which is inappropriate for these water tiles. Better choices would be `t_water_pool_outdoors` and `t_water_pool_shallow_outdoors` as these cannot be fished.

**Comment by actual-nh** [comment]
> Tip no 2: Salt and clean water can coexist IRL, google 'haloclines'

Salt and clean water as defined in the game could not coexist for any length of time, however; there would need to be a layer of (newly-defined) brackish (drinkable but with less quench and a possibility of nausea) water between them. (And, again, neither salt nor brackish water should be found in the water intended for a moose in a zoo.)

File: data/mods/DinoMod/mapgen/zoo.json [hunk 1]
```diff
@@ -385,7 +385,7 @@
         "7": "t_tree_dead",
         "B": "t_dirt",
         "D": "t_dirt",
-        "W": "t_water_sh",
+        "W": "t_water_dp",
```
Comment (actual-nh) [inline data/mods/DinoMod/mapgen/zoo.json:388]:
```suggestion
        "W": "t_water_pool_outdoors",
```

File: data/mods/DinoMod/mapgen/zoo.json [hunk 2]
```diff
@@ -395,7 +395,7 @@
         "f": "t_floor",
         "g": "t_chainfence_h",
         "h": "t_dirt",
-        "s": "t_swater_sh",
+        "s": "t_water_sh",
```
Comment (actual-nh) [inline data/mods/DinoMod/mapgen/zoo.json:398]:
```suggestion
        "s": "t_water_pool_shallow_outdoors",
```

File: data/mods/DinoMod/mapgen/zoo.json [hunk 3]
```diff
@@ -488,7 +488,7 @@
         "7": "t_tree_dead",
         "B": "t_dirt",
         "D": "t_dirt",
-        "W": "t_water_sh",
+        "W": "t_water_dp",
```
Comment (actual-nh) [inline data/mods/DinoMod/mapgen/zoo.json:491]:
```suggestion
        "W": "t_water_pool_outdoors",
```

File: data/mods/DinoMod/mapgen/zoo.json [hunk 4]
```diff
@@ -498,7 +498,7 @@
         "f": "t_floor",
         "g": "t_chainfence_h",
         "h": "t_dirt",
-        "s": "t_swater_sh",
+        "s": "t_water_sh",
```
Comment (actual-nh) [inline data/mods/DinoMod/mapgen/zoo.json:501]:
```suggestion
        "s": "t_water_pool_shallow_outdoors",
```

**Review by actual-nh — COMMENTED** [review]
This is, as best as I can tell, what was intended (deeper water surrounded by (glass or) shallow water.

File: data/mods/DinoMod/mapgen/zoo.json (hunk 2 above)
Comment (ps-19) [reply data/mods/DinoMod/mapgen/zoo.json]: Sir it is done now.

File: data/mods/DinoMod/mapgen/zoo.json (hunk 3 above)
Comment (ps-19) [reply data/mods/DinoMod/mapgen/zoo.json]: I made the change. Sir it is done now.

File: data/mods/DinoMod/mapgen/zoo.json (hunk 4 above)
Comment (ps-19) [reply data/mods/DinoMod/mapgen/zoo.json]: Ok

**Comment by ps-19** [comment]
> 
> 
> @ps-19
> 
> Three things:
> 
>     1. `Fixes issue #51265` in your summary here (under "purpose of change) should be changed to just `Fixes #51265` in order to have them linked so a merge of this PR closes the issue as well.
> 
>     2. As I mentioned in the issue, `t_water_dp` and `t_water_sh` are coded with the `"FISHABLE"` tag, which is inappropriate for these water tiles. Better choices would be `t_water_pool_outdoors` and `t_water_pool_shallow_outdoors` as these cannot be fished.
> 
>     3. The version you are fixing in this PR so far is only the version of the file under the DinoMod path (`Cataclysm-Dark Days Ahead\data\mods\DinoMod\mapgen`).
>        The base game version under `Cataclysm-Dark Days Ahead\data\json\mapgen` **_also_** needs these updates as well. You could do that as a separate PR, though you could also add changes to that file in this PR as well. It's up to you.

Thank you @RandDenner and @actual-nh for clearing mistakes in PR.

**Comment by RandDenner** [comment]
Looks good to me.
When you pass the checks, remember to go to the Discord and post it in the ready-to-merge channel.

**Review by LyleSY — APPROVED** [review]
Good call, thanks for catching this

**Pull request merged**